### DISCUSSION TITLE
Uptrace + Vector integration questions

### DISCUSSION THREAD

**Discussion by JohnBat26** [uptrace/uptrace#295] [Q&A] [answered]
Hello. I use 1.6.0-rc2 and try to set up vector & uptrace integration, and I have problems/questions:

1. Global search doesn't work, for example: `where {.name,.event_name} contains "error"`, though I send this fields from vector in json. Why?
2. There is `System` combobox  at overview screen, but I  don't understand how send this field from vector. 
3. Can I send metrics from vector?

**Comment by vmihailenco** [comment, marked as answer]
Hello,

>Global search doesn't work, for example: where {.name,.event_name} contains "error", though I send this fields from vector in json. Why?

You probably want `display.name contains error`. It is unlikely that `.name` and `.event_name` would contain `error`, but you can check with `group by .name, .event_name` what data you have there.

>There is System combobox at overview screen, but I don't understand how send this field from vector.

This field is set automatically set based on the span/event/log attributes. See [this](https://uptrace.dev/get/grouping.html#span-systems) for details.

>Can I send metrics from vector?

Nope, Vector metrics are not supported at the moment. We might support them in future, but for now I would recommend using [Otel Host Metrics](https://uptrace.dev/opentelemetry/collector-host-metrics.html). They are much better from what I can tell.

**Reply by JohnBat26** [reply]
@vmihailenco Can you give me full and correct example JSON with log event, where all fields filled out correctly:
  * with display.name
  * with event_name
  * with log attributed for system combobox

Unfortunately current version of uptrace documentation doesn't have complete examples for Vector+Uptrace integration

**Reply by vmihailenco** [reply]
@JohnBat26 it should work with the default Vector output. You can improve the end result by providing certain attributes, but it is not strictly necessary and it highly depends on the logs you're sending.

Please use the latest v1.6.0-rc.3 version (just pushed it) and create an issue if you think that logs are not handled correctly. You can view the data Vector sends to Uptrace using the [console](https://vector.dev/docs/reference/configuration/sinks/console/) sink. We have some information [here](https://uptrace.dev/get/ingest/vector.html).

You can also create a [free cloud account](https://app.uptrace.dev/join) to make it easier sharing the result with me so I can improve logs handling.

**Reply by JohnBat26** [reply]
I have installed v1.6.0-rc.3, but but I still don't see the logs displayed correctly in uptrace.
In uptrace log I see only:
```
uptrace-uptrace-1  | [bunrouter]  11:47:03.787  179a8c9efe74ad918fa03a6cd6a2efcb   200      1.134ms   POST     /api/v1/vector/logs
```

and queries logs, though I have next settings:
```yaml
logs:
  # Zap minimal logging level.
  # Valid values: DEBUG, INFO, WARN, ERROR, DPANIC, PANIC, FATAL.
  level: DEBUG

# Secret key that is used to sign JWT tokens etc.
secret_key: 102c1a557c314fc28198acd017960843

# Enable to log HTTP requests and database queries.
debug: true
```

Vector send next log event:
```
{
   "code":" 000b34",
   "deployment.environment":"stage",
   "file":"/opt/trueconf/server/var/log/vcs/stdout.log",
   "group_id":"STORAGE",
   "host":"tcs-asv-2",
   "log.errorcode":" 000b34",
   "log.file":"/opt/system/server/var/log/vcs/stdout.log",
   "log.message":"LDAP bind failed to demo.local:389: Can't contact LDAP server",
   "log.severity":"0",
   "log.source":"STORAGE",
   "log.timestamp":"2023-11-24T14:49:25.300Z",
   "message":"LDAP bind failed to demo.local:389: Can't contact LDAP server",
   "service.name":"STORAGE",
   "service.namespace":"tc-vcs",
   "source_type":"file",
   "span_kind":"server",
   "span_name":"STORAGE",
   "system":" STORAGE",
   "timestamp":"2023-11-24T14:49:25.300Z"
}
```
And Uptrace return `200 OK` (I see it from tcpdump)

**Questions:**
1. What JSON-fields am I setting incorrectly?
2. Why I can't see my log events in uptrace UI at all?
![image](https://github.com/uptrace/uptrace/assets/732942/1bf99773-2e87-439f-b281-dd821ccfc377)]]

**Reply by JohnBat26** [reply]
Syslogs evenst  I see in Uptrace, but don't see my events :(

**Reply by vmihailenco** [reply]
@JohnBat26 this date is in the future `"log.timestamp":"2023-11-24T14:49:25.300Z"` and that's why you don't see the log. Uptrace should probably ignore such dates...

**Reply by JohnBat26** [reply]
@vmihailenco Ок, I have fixed timezone, and now I see event in uptrace UI, but System combobox is empty and global search by `span_name` == `STORAGE` returns empty list

```json
{
   "code":" 000b05",
   "deployment.environment":"tc_stage",
   "file":"/opt/system/server/var/log/backend/error.log",
   "group_id":"STORAGE",
   "host":"tcs-asv-2",
   "log.errorcode":" 000b05",
   "log.file":"/opt/system/server/var/log/backend/error.log",
   "log.message":"Can't contact LDAP server",
   "log.severity":"0",
   "log.source":"STORAGE",
   "log.timestamp":"2023-11-24T16:42:58.998Z",
   "message":"Can't contact LDAP server",
   "service.name":"STORAGE",
   "service.namespace":"tc-backend",
   "source_type":"file",
   "span_kind":"server",
   "span_name":"STORAGE",
   "system":" STORAGE",
   "timestamp":"2023-11-24T16:42:58.998Z"
}
```

**Reply by JohnBat26** [reply]
@vmihailenco and Why I see `log.message` and `log.message1` though vector sent only `log.message1` ?

**Reply by JohnBat26** [reply]
Screen with `log.message1`

**Reply by JohnBat26** [reply]
https://disk.yandex.ru/i/Cnv0tT4BpzESkQ

**Reply by JohnBat26** [reply]
@vmihailenco And at Dashboard page: dashboard: `Uptrace: spans by service` is empty too?
how to create dashboard from logs?

**Reply by vmihailenco** [reply]
@JohnBat26 I am a bit busy at the moment and will check this a later before releasing the next version.
